src/utils/curve/curve.js

```diff
@@ -0,0 +1,29 @@
+import { retrieveData } from '../network'
+
+/**
+ * @function retrieve
+ * @memberof module:api.Curve
+ * @static
+ * @param {object} params - object of parameters:
+ * @param {number} params.AMId - Owning Asset Manager ID of the Profit and Loss record
+ * @param {Function} [callback] - Called with two arguments (error, result) on completion
+ * @returns {Promise|null} If no callback supplied, returns a Promise
+ */
+export function retrieve({ AMId, resourceId }, callback) {
+  const params = {
```

Comment:
As mentioned above, please refactor this to accept arguments that make sense in the context of use.